data/scripts/dependency-resolver.py

```diff
@@ -144,40 +205,125 @@ def handle_python_check(args, conf):
+        success = compile_test(source, args.compiler, "-Werror %s" % flags, None)
+        if success:
+            supported.append(i)
+
+    if supported:
+        context.add_append_makefile_var(append_to, " ".join(supported))
 
 type_handlers = {
     "pkg-config": handle_pkgconfig_check,
     "ccode": handle_ccode_check,
     "exec": handle_exec_check,
     "python": handle_python_check,
+    "cflags": handle_cflags_check,
 }
 
-def var_str(items):
+def format_makefile_var(items):
     output = ""
     for k,v in items:
-        if not v: continue
-        output += "%s ?= %s\n" % (k, v)
+        if not v or not v["value"]: continue
+        output += "%s %s %s\n" % (k, v["attrib"], v["value"])
     return output
 
-def makefile_gen(args):
-    output = ""
-    output += var_str(makefile_vars.items())
-    output += var_str(cfg_cflags.items())
-    output += var_str(cfg_ldflags.items())
+def makefile_gen(args, context):
+    output = format_makefile_var(context.get_makefile_vars().items())
     f = open(args.makefile_output, "w+")
     f.write(output)
     f.close()
 
-def kconfig_gen(args):
+def kconfig_gen(args, context):
     output = ""
-    for k,v in cfg_kconfig.items():
-        output += "config {config}\n{indent}bool\n{indent}default {enabled}\n". \
-                  format(config=k, indent="       ", enabled=v)
+    for k,v in context.get_kconfig().items():
+        output += "config {config}\n{indent}{ktype}\n{indent}default {enabled}\n". \
+                  format(config=k, indent="       ", ktype=v["type"], enabled=v["value"])
     f = open(args.kconfig_output, "w+")
     f.write(output)
     f.close()
 
+def run(args, dep_checks, context):
+    for i in dep_checks:
+        handler = type_handlers.get(i["type"])
+        if not handler:
+            print("Could not handle type: %s" % i["type"])
+            continue
+        handler(args, i, context)
+
+def vars_expand(origin, dest, maxrec):
+    remaining = {}
+
+    if not maxrec:
+        return
+
+    for k,v in origin.items():
+        if not isinstance(v, str):
+            dest[k] = v
+            continue
+
+        try:
+            dest[k] = v.format(**dest)
+        except KeyError:
+            remaining[k] = v
+
+    if remaining:
+        vars_expand(remaining, dest, maxrec - 1)
+
+def handle_definitions(args, conf, context):
+    definitions = conf.get("definitions")
+
+    variables = {"PREFIX": args.prefix}
+    vars_expand(definitions, variables, len(definitions))
+
+    header = ""
+    for k,v in variables.items():
+        if k == "PREFIX":
+            continue
+
+        if isinstance(v, str):
+            value = "\"%s\"" % v
+        else:
+            value = "%d" % v
+        context.add_cond_makefile_var(k, v)
+        header += "#define %s %s\n" % (k, value)
+
+    dirname = os.path.dirname(args.definitions_header)
+    if not os.path.exists(dirname):
+        os.makedirs(dirname)
+
+    f = open(args.definitions_header, "w")
```

Comment:
Minor, but you can  use the `with` syntax:

    with open(..., "w") as f:
        f.write(header)